src/general_manager/interface/readOnlyInterface.py

```
@@ -8,31 +8,77 @@
-        parent_class = getattr(cls, "_parent_class", None)
-        if model is None or parent_class is None:
-            raise ValueError("Attribute '_model' and '_parent_class' must be set.")
-        json_data = getattr(parent_class, "_json_data", None)
-        if not json_data:
+        if cls.ensureSchemaIsUpToDate(cls._parent_class, cls._model):
+            logger.warning(
+                f"Schema for ReadOnlyInterface '{cls._parent_class.__name__}' is not up to date."
+            )
+            return
+
+        model: Type[ReadOnlyModel] = getattr(cls, "_model")
```

Ruff (0.11.9)

Do not call `getattr` with a constant attribute value. It is not any safer than normal property access.

Replace `getattr` with attribute access

(B009)